catalog/rest/catalog-rest-endpoint/pom.xml

```diff
@@ -207,6 +213,7 @@
                             *
                         </Import-Package>
                         <Embed-Dependency>
+                            javax.activation,
```

Comment:
why do we embed this if it is only used in the test scope?

Reply:
I also thought it wasn't necessary, but without it as an embedded dependency, the bundle doesn't have what it needs when tests run and I get the following:

org.apache.felix.resolver.reason.ReasonException: Unable to resolve root: missing requirement [root] osgi.identity; osgi.identity=test-dependencies; type=karaf.feature; version="[0,0.0.0]"; filter:="(&(osgi.identity=test-dependencies)(type=karaf.feature)(version>=0.0.0)(version<=0.0.0))" [caused by: Unable to resolve test-dependencies/0.0.0: missing requirement [test-dependencies/0.0.0] osgi.identity; osgi.identity=catalog-rest-service-impl; type=osgi.bundle; version="[2.20.0.SNAPSHOT,2.20.0.SNAPSHOT]"; resolution:=mandatory [caused by: Unable to resolve catalog-rest-service-impl/2.20.0.SNAPSHOT: missing requirement [catalog-rest-service-impl/2.20.0.SNAPSHOT] osgi.wiring.package; filter:="(&(osgi.wiring.package=javax.activation)(version>=1.2.0)(!(version>=2.0.0)))"]]

The bundle doesn't have the required javax.activation package unless I define it as an embedded dependency, or at least Felix (or OSGI) can't resolve it unless it's there.

I'm fairly new to Karaf, OSGI, and Felix - so if there is an alternative method to resolve the bundle dependencies just for tests, let me know and I can try my hand at updating it!

Reply:
Ha I learned all about the basics of bundles and features, Karaf and Felix wiring, specifically because of this ticket. If nothing else, a great learning experience!

And I agree, this would likely be much, much cleaner if it was inside of a bundle and not polluting the pom. I'll give it a go!